src/Psl/Collection/SetInterface.php

```diff
@@ -0,0 +1,329 @@
+     * `SetInterface`.
+     *
+     * `$n` is 1-based. So the first element is 1, the second 2, etc.
+     *
+     * @param int<0, max> $n The last element to be skipped; the $n+1 element will be the
+     *                       first one in the returned `SetInterface`.
+     *
+     * @return SetInterface<T> A `SetInterface` that is a proper subset of the current
+     *                         `SetInterface` containing values after the specified `n`-th element.
+     *
+     * @psalm-mutation-free
+     */
+    public function drop(int $n): SetInterface;
+
+    /**
+     * Returns a `SetInterface` containing the values of the current `SetInterface`
+     * starting after and including the first value that produces `true` when
+     * passed to the specified callback.
+     *
+     * The returned `SetInterface` will always be a proper subset of the current
+     * `SetInterface`.
+     *
+     * @param (Closure(T): bool) $fn The callback used to determine the starting element for the
+     *                               returned `SetInterface`.
+     *
+     * @return SetInterface<T> A `SetInterface` that is a proper subset of the current
+     *                         `SetInterface` starting after the callback returns `true`.
+     */
+    public function dropWhile(Closure $fn): SetInterface;
+
+    /**
+     * Returns a subset of the current `SetInterface` starting from a given key up
+     * to, but not including, the element at the provided length from the starting
+     * key.
+     *
+     * `$start` is 0-based. $len is 1-based. So `slice(0, 2)` would return the
+     * elements at key 0 and 1.
+     *
+     * The returned `SetInterface` will always be a proper subset of this
+     * `SetInterface`.
+     *
+     * @param int<0, max> $start The starting key of this set to begin the returned
+     *                           `SetInterface`.
+     * @param int<0, max> $length The length of the returned `SetInterface`.
+     *
+     * @return SetInterface<T> A `SetInterface` that is a proper subset of the current
+     *                         `SetInterface` starting at `$start` up to but not including
+     *                         the element `$start + $length`.
+     *
+     * @psalm-mutation-free
+     */
+    public function slice(int $start, ?int $length = null): SetInterface;
+
+    /**
+     * Returns a `VectorInterface` containing the original `SetInterface` split into
+     * chunks of the given size.
+     *
+     * If the original `SetInterface` doesn't divide evenly, the final chunk will be
+     * smaller.
+     *
+     * @param positive-int $size The size of each chunk.
+     *
+     * @return VectorInterface<static<T>> A `VectorInterface` containing the original
+     *                                    `SetInterface` split into chunks of the given size.
+     *
+     * @psalm-mutation-free
+     */
+    public function chunk(int $size): VectorInterface;
+}
```

Comment:
Maybe some other usefull methods that could be added (or in another PR)

* diff
* union
* intersect